.gitignore

```diff
@@ -4,4 +4,5 @@ crytic-export/
 node_modules/
 typechain/
 foundry-out/
-.vscode/
\ No newline at end of file
+.vscode/
+solmate
```

Comment:
why do we have solmate top level anyways? it should be inside lib/ no?

Reply:
also it should be a submodule anyways so we don't want to gitignore it?

Reply:
ohh it was top level bc when I started this PR we still had hardhat (and for various reasons hardhat needs it top level lol). Deleting!

Reply:
When its top level its still a submodule in `lib` but then you do a symlink at the top level so hardhat can find it